File: Level2/구명보트.py
people = [70, 50, 80]
limit = 100


def solution(people, limit):
    people.sort()
    count = 0
    answer = 0
    index = len(people) - 1

    while index >= count:
        tmp = people[index]
        index -= 1
        if people[count] + tmp <= limit:
            count += 1
            answer += 1
        else:
            answer += 1
    return answer






did = solution(people, limit)
print(did)

File: Level2/구명보트.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
	int answer = 0;
	sort(people.begin(), people.end()); // 오름차순 정렬
	int count = 0;

	while (people.size() > count)
	{
		int tmp = people.back();
		people.pop_back();
		//제일 큰 값 + 제일 작은 값 조합을 계속 찾아나감(greedy)
		if (tmp + people[count] <= limit)
		{
			++answer;
			++count;
		}
		else //둘 중 한 명만 태우는 경우
			++answer;
	}
	return answer;
}